src/models/MSC3089TreeSpace.ts

```diff
@@ -0,0 +1,433 @@
+            via: [this.client.getDomain()],
+        }, this.roomId);
+
+        return directory;
+    }
+
+    /**
+     * Gets a list of all known immediate subdirectories to this tree space.
+     * @returns {MSC3089TreeSpace[]} The tree spaces (directories). May be empty, but not null.
+     */
+    public getDirectories(): MSC3089TreeSpace[] {
+        const trees: MSC3089TreeSpace[] = [];
+        const children = this.room.currentState.getStateEvents(EventType.SpaceChild);
+        for (const child of children) {
+            try {
+                const tree = this.client.unstableGetFileTreeSpace(child.getStateKey());
+                if (tree) trees.push(tree);
+            } catch (e) {
+                logger.warn("Unable to create tree space instance for listing. Are we joined?", e);
+            }
+        }
+        return trees;
+    }
+
+    /**
+     * Gets a subdirectory of a given ID under this tree space. Note that this will not recurse
+     * into children and instead only look one level deep.
+     * @param {string} roomId The room ID (directory ID) to find.
+     * @returns {MSC3089TreeSpace} The directory, or falsy if not found.
+     */
+    public getDirectory(roomId: string): MSC3089TreeSpace {
+        return this.getDirectories().find(r => r.roomId === roomId);
+    }
+
+    /**
+     * Deletes the tree, kicking all members and deleting **all subdirectories**.
+     * @returns {Promise<void>} Resolves when complete.
+     */
+    public async delete(): Promise<void> {
+        const subdirectories = this.getDirectories();
+        for (const dir of subdirectories) {
+            await dir.delete();
+        }
+
+        const kickMemberships = ["invite", "knock", "join"];
+        const members = this.room.currentState.getStateEvents(EventType.RoomMember);
+        for (const member of members) {
+            const isNotUs = member.getStateKey() !== this.client.getUserId();
+            if (isNotUs && kickMemberships.includes(member.getContent()['membership'])) {
+                await this.client.kick(this.roomId, member.getStateKey(), "Room deleted");
+            }
+        }
+
+        await this.client.leave(this.roomId);
+    }
+
+    private getOrderedChildren(children: MatrixEvent[]): { roomId: string, order: string }[] {
+        const ordered: { roomId: string, order: string }[] = children
+            .map(c => ({ roomId: c.getStateKey(), order: c.getContent()['order'] }));
+        ordered.sort((a, b) => {
+            if (a.order && !b.order) {
+                return -1;
+            } else if (!a.order && b.order) {
+                return 1;
+            } else if (!a.order && !b.order) {
+                const roomA = this.client.getRoom(a.roomId);
+                const roomB = this.client.getRoom(b.roomId);
+                if (!roomA || !roomB) { // just don't bother trying to do more partial sorting
+                    return a.roomId.localeCompare(b.roomId);
```

Comment:
```suggestion
                    return compare(a.roomId, b.roomId);
```

To import from `src/utils/strings`. They do pretty much the same thing but reusing a `Intl.Collator` object is improving performance tremendously

Reply:
This is js-sdk, your string util is in react-sdk, maybe it ought to be moved.